tracing/trace.go

```diff
@@ -49,16 +51,36 @@ type trace struct {
 	tags     map[string]string
 }
 
+func (t *trace) getTracer() *Tracer {
+	if t == nil {
+		return nil
+	}
+	if tracer, ok := t.tracer.(*Tracer); ok && tracer != nil {
+		return tracer
+	}
+	return nil
+}
+
 func newTrace(tracer *Tracer, name string) *trace {
+	var (
+		otTracer        opentracing.Tracer
+		traceID, spanID string
+	)
 	if tracer == nil {
-		tracer = &globalTracer
+		otTracer = opentracing.GlobalTracer()
+		traceID = globalTracer.newTraceID()
+		spanID = globalTracer.newSpanID()
```

Comment:
🤔 this seems wrong -- shouldn't it be `otTracer.NewTraceID()`? in which case can't these go after the if statement?  If not, then something subtle is happening here and we'll want a comment to explain

Reply:
`opentracing.Tracer` is an interface that has only `StartSpan`, `Inject`, and `Extract` methods. This function expects a `*tracing.Tracer`, but the global `opentracing.Tracer` may not be one. That's why it's relying on the `globalTracer`. I wrote it this way so that even if the `opentracing.GlobalTracer()` has been overridden, the IDs generated by baseplate will still conform to the specified configuration.

Reply:
Definitely worth a comment, then!